packages/graphql-docs-generator/src/generator/getFields.ts

```diff
@@ -40,6 +40,17 @@ export default function getFields(
       return getFields(subField, schema, adjustDepth(subField, depth), options);
     })
     .filter(f => f);
+
+  // add __typename to selection set.
+  // getFields() does not include __typename because __typename is implicitly included on all object types.
+  if (isObjectType(fieldType)) {
```

Comment:
Do Interfaces also have this meta field? If yes we might need to modify this to `isObjectType(fieldType) || isInterfaceType(fieldType)`.